arcgis-ios-sdk-samples/Route and directions/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -0,0 +1,151 @@
+        resetBarButtonItem.isEnabled = true
+        // Set NMEA location data source for location display.
+        mapView.locationDisplay.dataSource = nmeaLocationDataSource
+        // Start the data source and location display.
+        mockNMEADataSource.start()
+        mapView.locationDisplay.start()
+        // Recenter the map and set pan mode.
+        recenter()
+    }
+    
+    @IBAction func recenter() {
+        mapView.locationDisplay.autoPanMode = .recenter
+        recenterBarButtonItem.isEnabled = false
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
+            DispatchQueue.main.async { [weak self] in
+                self?.recenterBarButtonItem.isEnabled = true
+            }
+            self?.mapView.locationDisplay.autoPanModeChangedHandler = nil
+        }
+    }
+    
+    @IBAction func reset() {
+        // Reset buttons states.
+        startBarButtonItem.isEnabled = true
+        resetBarButtonItem.isEnabled = false
+        // Reset the status text.
+        statusLabel.text = "Satellites info will be shown here."
+        // Reset and stop the location display.
+        mapView.locationDisplay.autoPanModeChangedHandler = nil
+        mapView.locationDisplay.autoPanMode = .off
+        mapView.locationDisplay.stop()
+        // Pause the mock data generation.
+        mockNMEADataSource.stop()
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "DisplayDeviceLocationWithNMEADataSourcesViewController",
+            "SimulatedNMEADataSource"
+        ]
+        // Load NMEA location data source.
+        startBarButtonItem.isEnabled = true
+        nmeaLocationDataSource.locationChangeHandlerDelegate = self
+        mockNMEADataSource.delegate = self
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        reset()
+    }
+}
+
+// MARK: SimulatedNMEADataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: SimulatedNMEADataSourceDelegate {
+    func dataSource(didUpdate nmeaData: Data) {
+        // Push mock data into the data source.
+        // Note: You can also get real-time NMEA sentences from a GPS dongle.
+        nmeaLocationDataSource.push(nmeaData)
+    }
+}
+
+// MARK: AGSNMEALocationDataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: AGSNMEALocationDataSourceDelegate {
+    func nmeaLocationDataSource(_ NMEALocationDataSource: AGSNMEALocationDataSource, satellitesDidChange satellites: [AGSNMEASatelliteInfo]) {
+        // Update the satellites info status text.
+        let satelliteSystemsText = Set(satellites.map { $0.system }).map { $0.label }.joined(separator: ", ")
```

Comment:
The end result of the second map is an array of unique strings. Instead of mapping twice, I think you could combine the two to just map once. Also, it would be a good idea to sort the strings so that you don't end up with multiple list items flip flopping.

```suggestion
        let satelliteSystemsText = Set(satellites.map(\.system.label))
            .sorted()
            .joined(separator: ", ")
```

I wish we could use `ListFormatter`. You could, but it isn't available on iOS 12, so you'd still need to use `joined(separator:)` for that.

Reply:
BTW I love the backslashdot-map here! 🤩 
Now I start to wonder if I could've used that in those previous samples...